packages/core/elasticsearch/core-elasticsearch-client-server-internal/src/log_query_and_deprecation.ts

```diff
@@ -13,8 +13,59 @@ import { errors, DiagnosticResult, RequestBody, Client } from '@elastic/elastics
 import numeral from '@elastic/numeral';
 import type { Logger } from '@kbn/logging';
 import type { ElasticsearchErrorDetails } from '@kbn/es-errors';
+import type { ElasticsearchPotentiallyLeakingApi } from '@kbn/core-elasticsearch-server';
 import { getEcsResponseLog } from './get_ecs_response_log';
 
+/**
+ * The logger-relevant request meta of an ES request
+ */
+export interface RequestDebugMeta {
+  /**
+   * The requested method
+   */
+  method: string;
+  /**
+   * The requested endpoint + querystring
+   */
+  url: string;
+  /**
+   * The request body (it may be redacted)
+   */
+  body: string;
+  /**
+   * The status code of the response
+   */
+  statusCode: number | null;
+}
+
+const POTENTIALLY_LEAKING_API: ElasticsearchPotentiallyLeakingApi[] = [
+  { path: '/_security/' },
+  { path: '/_xpack/security/' },
+  { method: 'POST', path: '/_reindex' },
+  { method: 'PUT', path: '/_watcher/watch' },
+  { method: 'PUT', path: '/_xpack/watcher/watch' },
+  { method: 'PUT', path: '/_snapshot/' },
+  { method: 'PUT', path: '/_logstash/pipeline/' },
+  { method: 'POST', path: '/_nodes/reload_secure_settings' },
+  { method: 'POST', path: /\/_nodes\/.+\/reload_secure_settings/ },
```

Comment:
According to the docs it is: https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-nodes-reload-secure-settings.html

The reason I added the method is to avoid matching regexp to all requests.